vortex-array/src/arrays/list/mod.rs

```diff
@@ -100,40 +101,58 @@ impl ListArray {
         })
     }
 
-    // TODO: merge logic with varbin
-    // TODO(ngates): should return a result if it requires canonicalizing offsets
-    pub fn offset_at(&self, index: usize) -> usize {
```

Comment:
this feels like a weird use of result

Reply:
Indeed, we moved the varbin to return result so I made this match. Can swap both tbh